events/listener.go

```diff
@@ -139,11 +145,17 @@ func (w *Worker) DoWork(rawEvent []byte, eventHandlers map[string]EventHandler,
 	if fn, ok := eventHandlers[event.Name]; ok {
 		err = fn(event, apiClient)
 		if err != nil {
-			log.Printf("Error processing event. Event name: %v. Event id: %v Resource id: %v. Error: %v",
-				event.Name, event.Id, event.ResourceId, err)
+			log.WithFields(log.Fields{
+				"EventName":  event.Name,
+				"EventId":    event.Id,
+				"ResourceId": event.ResourceId,
+				"Err":        err,
+			}).Warn("Error processing event")
```

Comment:
This was the other place I suggested error level but github hid the original comment when you changed the field names to camel case.

I stand by making this an error level log as well. This error represents the errors raised by the handlers. If a handler is raising an error, that means rancher is failing to deploy a machine. I think it is reasonable for us to log this as an error and that Ops guys would want to see if machines were failing to deploy.

That said, the other thing I ultimately want to do here is update the transitioningMessage on machine api resource with the error so that the user can know how/why the machine errored out.